src/app/project/controllers/collect-records.ctrl.js

```diff
@@ -466,6 +526,27 @@ angular.module('app.project').controller('CollectRecordsCtrl', [
       });
     }
 
+    function selectAllOptions(allSelected, filter_options) {
+      var filterTypes = filter_options.filterTypes;
+      var choices = filter_options.choices;
+      var storageName = filter_options.storageName;
+      var options = JSON.parse(localStorage.getItem(storageName)) || choices;
+
+      if (allSelected) {
+        filterTypes.forEach(filterItem => {
+          if (!filterItem.selected) {
```

Comment:
Do we need the conditional here?

Reply:
yeah, only push the ones with the checkbox not selected to local storage